### PR TITLE
Removing YAML Config Files For Cobra Binaries

### PR DESCRIPTION
**Pull request by Hirdey-1999**
### Changes to be committed:
	modified:   cmd/apprepository-controller/cmd/root.go
	modified:   cmd/asset-syncer/cmd/root.go
	modified:   cmd/kubeapps-apis/cmd/root.go

### Description of the change
Made Changes To The :
1. cmd/apprepository-controller/cmd/root.go
2. cmd/asset-syncer/cmd/root.go
3. cmd/kubeapps-apis/cmd/root.go

### Benefits
This Helped In The Removing Of YAML Config Files For Cobra Binaries

### Additional information

[user]
	email = hirdeymittal@gmail.com
	name = hirdey-1999
	signingkey = 3B89C963E92E9EE143862550D210328360EE7B34
[commit]
	gpgsign = true
[gpg]
	program = gpg

### CODE REVIEW

File: cmd/kubeapps-apis/cmd/root.go [hunk 1]
```diff
@@ -82,13 +82,7 @@ func initConfig() {
 		// Use config file from the flag.
 		viper.SetConfigFile(cfgFile)
 	} else {
```
Comment (castelblanque) [inline cmd/kubeapps-apis/cmd/root.go:84]:
```suggestion
	}
```

File: cmd/kubeapps-apis/cmd/root.go [hunk 2]
```diff
@@ -82,13 +82,7 @@ func initConfig() {
...
-		// Find home directory.
-		home, err := homedir.Dir()
-		cobra.CheckErr(err)
 
-		// Search config in home directory with name ".kubeapps-apis" (without extension).
-		viper.AddConfigPath(home)
-		viper.SetConfigName(".kubeapps-apis")
 	}
```
Comment (castelblanque) [inline cmd/kubeapps-apis/cmd/root.go:86]:
```suggestion
```

File: cmd/asset-syncer/cmd/root.go [hunk 3]
```diff
@@ -142,14 +142,7 @@ func initConfig() {
 		// Use config file from the flag.
 		viper.SetConfigFile(cfgFile)
 	} else {
```
Comment (castelblanque) [inline cmd/asset-syncer/cmd/root.go:144]:
```suggestion
	}
```

File: cmd/asset-syncer/cmd/root.go [hunk 4]
```diff
@@ -142,14 +142,7 @@ func initConfig() {
...
-		// Find home directory.
-		home, err := homedir.Dir()
-		cobra.CheckErr(err)
-
-		// Search config in home directory with name ".kubeops" (without extension).
-		viper.AddConfigPath(home)
-		viper.SetConfigType("yaml")
-		viper.SetConfigName(".kubeops")
+
 	}
```
Comment (castelblanque) [inline cmd/asset-syncer/cmd/root.go:146]:
```suggestion
```

File: cmd/apprepository-controller/cmd/root.go [hunk 5]
```diff
@@ -101,14 +101,7 @@ func initConfig() {
 		// Use config file from the flag.
 		viper.SetConfigFile(cfgFile)
 	} else {
```
Comment (castelblanque) [inline cmd/apprepository-controller/cmd/root.go:103]:
```suggestion
	}
```

File: cmd/apprepository-controller/cmd/root.go [hunk 6]
```diff
@@ -101,14 +101,7 @@ func initConfig() {
...
-		// Find home directory.
-		home, err := homedir.Dir()
-		cobra.CheckErr(err)
-
-		// Search config in home directory with name ".kubeops" (without extension).
-		viper.AddConfigPath(home)
-		viper.SetConfigType("yaml")
-		viper.SetConfigName(".kubeops")
+		
 	}
```
Comment (castelblanque) [inline cmd/apprepository-controller/cmd/root.go:105]:
```suggestion
```

File: .gitpod.yml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# This configuration file was automatically generated by Gitpod.
+# Please adjust to your needs (see https://www.gitpod.io/docs/config-gitpod-file)
+# and commit this file to your remote git repository to share the goodness with others.
+
+tasks:
+  - init: go get && go build ./... && go test ./... && make
+    command: go run
+
+
```
Comment (castelblanque) [inline .gitpod.yml:9]: Usually local development tools files are not merged in the repo.
It is just a matter of adding `.gitpod.yml` to the [.gitignore file](https://github.com/vmware-tanzu/kubeapps/blob/main/.gitignore).

**Comment by castelblanque** [comment]
@Hirdey-1999 Many thanks for the contribution to the community, really appreciated!
Just a couple of comments from my side, as you can see in GH: add the `gitpod.yml` file to .gitignore and completely remove the empty `else` branch.
Thanks again!

**Comment by Hirdey-1999** [comment]
@castelblanque Made With The Required Changes Kindly Check With The Code Let Me Know With Changes 
Very Thanks @castelblanque

**Comment by castelblanque** [comment]
Many thanks for the promptness @Hirdey-1999 , looks good to me!

**Review by castelblanque — APPROVED** [review]

**Comment by Hirdey-1999** [comment]
@antgamdia Very Thanks Made A Change To The Title Of PR Let Me Know With Any Changes Very Thankful To The Team

**Comment by castelblanque** [comment]
hi @Hirdey-1999 , sorry, but GitHub won't allow to merge the PR with unsigned commits.

There are two requirements that need to be met:
- Sign off commits with the line `Signed-off-by: <Your-Name> <Your-email>`. [Related GH blogpost with info.](https://github.blog/changelog/2022-06-08-admins-can-require-sign-off-on-web-based-commits/)
- Sign commits (it is different from the above). [More info](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits). Please generate a PGP key and use it for signing your commits.

Probably a good approach is to create a new branch from main, apply the changes and create a commit that meets the two requirements above. Then a new PR can be created.

Sorry for the inconvenience, we will update [Contributing documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) to warn about this requirements.

Please let me know if you need help for the process, happy to have a chat in [Slack](https://kubernetes.slack.com).

**Comment by Hirdey-1999** [comment]
Happy To Help Again

**Comment by Hirdey-1999** [comment]
I am facing issue with signing with commit need help

**Comment by castelblanque** [comment]
hi @Hirdey-1999 , happy to help. Please ping me on [Slack channel](https://kubernetes.slack.com/archives/C9D3TSUG4) and let me know what time slot suits you best.

**Pull request closed without merging**